packages/trie/src/baseTrie.ts

```diff
@@ -561,16 +569,22 @@ export class Trie {
   ): Buffer | (EmbeddedNode | null)[] {
     const rlpNode = node.serialize()
 
+    // TODO: analyze why this length check is here, 2021-04-29
+    // If removed various tests fail. However this prevents certain delete operations
+    // when deleteFromDB is activated
+    // Related test: `index.spec.ts`, "setting back state root (deleteFromDB)"
```

Comment:
Copying this answer from my prior comment, on [eth.wiki](https://eth.wiki/fundamentals/patricia-tree) it says:
> When one node is referenced inside another node, what is included is `H(rlp.encode(x))`, where `H(x) = sha3(x) if len(x) >= 32 else x` and `rlp.encode` is the RLP encoding function.